src/platform/logging/impl/stdio/Logging.cpp

```diff
@@ -4,17 +4,64 @@
 
 #include <lib/support/logging/Constants.h>
 
+#include <inttypes.h>
 #include <stdio.h>
+#include <time.h>
+
+#if defined(__APPLE__)
+#include <pthread.h>
+#include <unistd.h>
+#elif defined(__gnu_linux__)
+#include <sys/syscall.h>
+#include <unistd.h>
+#endif
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
+// TODO: investigate whether pw_chrono or pw_log could be used here
 void LogV(const char * module, uint8_t category, const char * msg, va_list v)
 {
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
+    flockfile(stdout);
+#endif
+
+    switch (category)
+    {
+    case kLogCategory_Error:
+        printf("\033[1;31m");
+        break;
+    case kLogCategory_Progress:
+        printf("\033[0;32m");
+        break;
+    case kLogCategory_Detail:
+        printf("\033[0;34m");
+        break;
+    }
+
+#if defined(__APPLE__) || defined(__gnu_linux__)
+    timespec ts;
+    timespec_get(&ts, TIME_UTC);
+    printf("[%lld.%06ld] ", static_cast<long long>(ts.tv_sec), static_cast<long>(ts.tv_nsec / 1000));
+#endif
+
+#if defined(__APPLE__)
+    uint64_t ktid;
+    pthread_threadid_np(nullptr, &ktid);
+    printf("[%lld:%lld] ", static_cast<long long>(getpid()), static_cast<long long>(ktid));
+#elif defined(__gnu_linux__) && !defined(__NuttX__)
+    // TODO: change to getpid() and gettid() after glib upgrade
+    printf("[%lld:%lld] ", static_cast<long long>(syscall(SYS_getpid)), static_cast<long long>(syscall(SYS_gettid)));
+#endif
+
     printf("CHIP:%s: ", module);
```

Comment:
This is also changing the formatting in the Apple case.  Again, probably fine, but we'll need to watch out for issues in log processing tooling.

Why do we want to include "CHIP" in here?

Reply:
I had based my format string on the linux logger:

https://github.com/project-chip/connectedhomeip/blob/58536ba64da75f0a63c569baa29966008ffdf306/src/platform/Linux/Logging.cpp#L51-L52

But you're right, it's better to stay consistent with darwin. So, I changed it; now a log line looks like this:
```
[1719389130094] [1459444:1459446] [IN] TransportMgr initialized
```

Reply:
> But you're right, it's better to stay consistent with darwin.

But now it's inconsistent with Linux :D It's not possible to keep backward compatibility in case of unification.

I agree that the `CHIP` prefix should go away. It could make sense in case of system logger where logs from many applications are redirected to a single file, but it's not the case for STDIO I think. But I'm not sure about the time format. The dot between seconds and the rest is really useful for humans. Also, millisecond precision will not show time changes in many cases (microsecond precision does).

Reply:
I agree that unless we have more ifdefs one or the other log format will change.  I'm just saying we should watch out for issues getting filed because someone's tools broke, so we can tell them what the story is.

Regarding time format, in practice, I suspect that seconds.milliseconds is the most useful time format here, in terms of readability, if we posit that it's humans reading the output.  It probably does not matter in most cases how much time has passed between two log lines if it's < 1ms.